scripts/devel/generate_devel_script.py

```diff
@@ -80,6 +89,23 @@ def beforeInclude(self, *_, **kwargs):
                     script = script.replace(
                         ' --build-tool catkin_make_isolated',
                         ' --build-tool ' + args.build_tool)
+                if args.build_tool_args is not None or args.build_tool_test_args is not None:
+                    lines = script.splitlines()
+                    for i, line in enumerate(lines):
```

Comment:
I'm not proud of this. Any better suggestions?

Reply:
Now I understand. You are trying to offer command line arguments to *override* the build and test args. I am not sure that is necessary. I don't have a better suggestion atm - it can stay as is.

The args might also need to escape double quotes here though.

Reply:
In the CI jobs, the command line arguments replace the arguments from the build file, but it is a lot more straightforward because they are already Jenkins parameters.

Would it be a good idea to just do the same thing here, and expose these parameters in Jenkins? Do we ever manually trigger devel/pr jobs?

Would you rather see the CI jobs change to be additive? I'd strongly prefer that the behavior was the same, but I don't have a ton of feeling about which is preferable.

Reply:
> Do we ever manually trigger pr jobs?

No.

> Do we ever manually trigger devel jobs?

Yes, but without custom parameters. E.g. to check if the status changed due to upstream package changes even though the tested repo hasn't changed.

> Would you rather see the CI jobs change to be additive? I'd strongly prefer that the behavior was the same, but I don't have a ton of feeling about which is preferable.

Atm we don't have the option and I am not sure we even have a use case where we would want to pass custom args. So I don't mind what to do. It would be good though if the user calling the CLI has an idea (maybe through `--help`) what the default args are.

Reply:
> It would be good though if the user calling the CLI has an idea what the default args are.

As for printing the default args in the help text, I'm really not sure how to implement this cleanly because we need to already know what ros distro was specified on the command line in order to get the build config. If you have any suggestions on how to accomplish this, I'd appreciate it.

Reply:
That would only work if the user passes the necessary arguments and uses `--help` after that. Then the code could parse known args and populate the help text accordingly. Not really a great approach but I don't have any other idea.